packages/app-folders/src/components/Tree/index.tsx

```diff
@@ -79,22 +82,29 @@ type Props = {
     type: string;
     title: string;
     onFolderClick: (data: NodeModel<DndItemData>["data"]) => void;
+    onTitleClick?: (event: React.MouseEvent<HTMLElement>) => void;
     focusedFolderId?: string;
 };
 
-export const FolderTree: React.FC<Props> = ({ type, title, focusedFolderId, onFolderClick }) => {
+export const FolderTree: React.FC<Props> = ({
+    type,
+    title,
+    focusedFolderId,
+    onFolderClick,
+    onTitleClick
+}) => {
     const { folders, updateFolder } = useFolders(type);
     const [treeData, setTreeData] = useState<NodeModel<DndItemData>[]>([]);
     const [createDialogOpen, setCreateDialogOpen] = useState<boolean>(false);
     const [initialOpenList, setInitialOpenList] = useState<undefined | InitialOpen>(undefined);
     const { showSnackbar } = useSnackbar();
 
-    useEffect(() => {
+    useDeepCompareEffect(() => {
         if (folders) {
             setTreeData(createTreeData(folders, focusedFolderId));
             setInitialOpenList(createInitialOpenList(folders, focusedFolderId));
         }
-    }, [folders]);
+    }, [Object.assign({}, folders), focusedFolderId]);
```

Comment:
This `Object.assign` looks problematic to me, because it creates a new object on every render, so even the internal `ref` can't help, because it will always get a new object. Am I wrong?

Why not just send `folders` in there?

Reply:
Sending the folders array won't work, that's the reason why I use this workaround